ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -77,91 +82,62 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
+    awsConfig.partSize = partSize;
 
     if ((fs = (S3_FILE_SYSTEM *)calloc(1, sizeof(S3_FILE_SYSTEM))) == NULL)
         return (errno);
 
     fs->s3Storage = (S3_STORAGE *)storageSource;
 
     /* New can fail; will deal with this later. */
-    fs->conn = new S3Connection(aws_config);
-
+    fs->conn = new S3Connection(awsConfig);
+    fs->fileSystem.fs_directory_list = S3DirectoryList;
+    fs->fileSystem.fs_directory_list_single = S3DirectoryListSingle;
     fs->fileSystem.terminate = S3FileSystemTerminate;
 
     /* TODO: Move these into tests. Just testing here temporarily to show all functions work. */
     {
-        /* List S3 buckets. */
         std::vector<std::string> buckets;
-        if (fs->conn->ListBuckets(buckets)) {
-            std::cout << "All buckets under my account:" << std::endl;
-            for (const std::string &bucket : buckets) {
-                std::cout << "  * " << bucket << std::endl;
-            }
-            std::cout << std::endl;
-        }
+        fs->conn->ListBuckets(buckets);
+        std::cout << "All buckets under my account:" << std::endl;
+        for (const std::string &bucket : buckets)
+            std::cout << "  * " << bucket << std::endl;
+        std::cout << std::endl;
 
         /* Have at least one bucket to use. */
         if (!buckets.empty()) {
-            const Aws::String firstBucket = buckets.at(0);
-
-            /* List objects. */
-            std::vector<std::string> bucketObjects;
-            if (fs->conn->ListObjects(firstBucket, bucketObjects)) {
-                std::cout << "Objects in bucket '" << firstBucket << "':" << std::endl;
-                if (!bucketObjects.empty()) {
-                    for (const auto &object : bucketObjects) {
-                        std::cout << "  * " << object << std::endl;
-                    }
-                } else {
-                    std::cout << "No objects in bucket." << std::endl;
-                }
-                std::cout << std::endl;
-            }
+            const std::string firstBucket = buckets.at(0);
 
             /* Put object. */
             fs->conn->PutObject(firstBucket, "WiredTiger.turtle", "WiredTiger.turtle");
 
-            /* List objects again. */
-            bucketObjects.clear();
-            if (fs->conn->ListObjects(firstBucket, bucketObjects)) {
-                std::cout << "Objects in bucket '" << firstBucket << "':" << std::endl;
-                if (!bucketObjects.empty()) {
-                    for (const auto &object : bucketObjects) {
-                        std::cout << "  * " << object << std::endl;
-                    }
-                } else {
-                    std::cout << "No objects in bucket." << std::endl;
-                }
-                std::cout << std::endl;
-            }
+            /* Testing directory list. */
+            WT_SESSION *session = NULL;
+            const char *directory = firstBucket.c_str();
```

Comment:
Can we use `string` here instead and use `c_str()` wherever we expect `const char *`?